addons/account/models/account_account.py

```diff
@@ -622,13 +622,14 @@ def _adapt_parent_account_group(self):
                 AND parent.code_prefix_start <= LEFT(agroup.code_prefix_start, char_length(parent.code_prefix_start))
                 AND parent.code_prefix_end >= LEFT(agroup.code_prefix_end, char_length(parent.code_prefix_end))
                 AND parent.id != agroup.id
-                AND parent.company_id = %(company_id)s
+                AND parent.company_id = agroup.company_id
+                AND parent.company_id in %(company_ids)s
```

Comment:
Since `parent.company_id = agroup.company_id` and `agroup.company_id in %(company_ids)s`, it is redundant to put `parent.company_id in %(company_ids)s`

Reply:
There is most certainly a way to rewrite this query to avoid doing a sub-select; might be the right time to do it.

It might also be worth to flush only `account.group` above with 
```python
self.env['account.group'].flush(self.env['account.group']._fields)
```

Reply:
to avoid doing a sub-select I am stuck in ORDER BY and LIMIT 1.
can you plz help?